src/org/labkey/test/AssayAPITest.java

```diff
@@ -106,6 +111,40 @@ protected void importAssayAndRun(File assayPath, int pipelineCount, String assay
         assertTextPresent(textToCheck);
     }
 
+    @Test
+    public void testGpatAssayOverAPI() throws Exception
+    {
+        String assayName = "testGpatAssay";
+        String assayDescription = "generated for test purposes over remoteAPI";
+
+        Connection connection = createDefaultConnection(true);
+        GetProtocolCommand getProtocolCommand = new GetProtocolCommand("General");                      // gets a template from the server
+        ProtocolResponse getProtocolResponse = getProtocolCommand.execute(connection, getCurrentContainerPath());
+
+        Protocol newAssayProtocol = getProtocolResponse.getProtocol();
+        newAssayProtocol.setName(assayName)
+            .setDescription(assayDescription)
+            .allowQCStates(true)
+            .setEditableResults(true)
+            .setEditableRuns(true);
+        SaveProtocolCommand saveProtocolCommand = new SaveProtocolCommand(newAssayProtocol);
+        ProtocolResponse saveProtocolResponse = saveProtocolCommand.execute(connection, getCurrentContainerPath());
+        Long protocolId = saveProtocolResponse.getProtocol().getProtocolId();
+
+        assertEquals(assayDescription, saveProtocolResponse.getProtocol().getDescription());
+        assertTrue(saveProtocolResponse.getProtocol().getAllowQCStates());
+        assertTrue(saveProtocolResponse.getProtocol().getEditableResults());
+        assertTrue(saveProtocolResponse.getProtocol().getEditableRuns());
+
+        GetProtocolCommand protocolCommand = new GetProtocolCommand(protocolId);
+        ProtocolResponse doubleCheckProtocolResponse = protocolCommand.execute(connection, getCurrentContainerPath());
+
+        assertEquals(assayDescription, doubleCheckProtocolResponse.getProtocol().getDescription());
+        assertTrue(doubleCheckProtocolResponse.getProtocol().getAllowQCStates());
```

Comment:
Shouldn't getAllowQCStates() be getQcEnabled()?

Reply:
not unless allowing qc states is the same as setting them enabled/disabled